component-catalog-connectors/example-components-connector/elyra_examples_connector/examples_connector.py

```diff
@@ -0,0 +1,117 @@
+#
+# Copyright 2018-2021 Elyra Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from pathlib import Path
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+
+from elyra.pipeline.catalog_connector import ComponentCatalogConnector
+
+
+class ExamplesCatalogConnector(ComponentCatalogConnector):
+    """
+    Makes example components for Kubeflow Pipelines and Apache Airflow
+    available to Elyra.
+    """
+    config = {
+        'kfp': {
+            'root_dir': 'kfp_example_components',
+            'file_filter': '*.yaml'
+        },
+        'airflow': {
+            'root_dir': 'airflow_example_components',
+            'file_filter': '*.py'
+        }
+    }
+
+    def get_catalog_entries(self, catalog_metadata: Dict[str, Any]) -> List[Dict[str, Any]]:
+        """
+        Returns Elyra example custom components for the selected runtime (Kubeflow Pipelines
+        and Apache Airflow only)
+        :param registry_metadata: the dictionary-form of the Metadata instance for a single registry
+        """
+        component_list = []
+
+        runtime_id = catalog_metadata.get('runtime')
+
+        if runtime_id not in ExamplesCatalogConnector.config.keys():
+            self.log.error(f'Cannot retrieve component list for runtime \'{runtime_id}\': '
+                           f'only {list(ExamplesCatalogConnector.config.keys())} are supported.')
+            # return empty component specification list
+            return component_list
+
+        try:
+            root_dir = Path(__file__).parent / ExamplesCatalogConnector.config[runtime_id]['root_dir']
+            self.log.debug(f"Retrieving component list for runtime '{catalog_metadata.get('runtime')}' from "
+                           f'{root_dir}')
+            pattern = ExamplesCatalogConnector.config[runtime_id].get('file_filter', '*')
+            self.log.debug(f'Component file pattern: {pattern}')
+            for file in root_dir.glob(f'**/{pattern}'):
+                component_list.append({'component-id': str(file)[len(str(root_dir)) + 1:]})
```

Comment:
Yep - that was the intention. Since this connector serves multiple purposes (delivers example components + blueprint for other connectors) I wanted to keep the key name generic.